common/cards/alter-egos/effects/turtle-shell.ts

```diff
@@ -23,29 +23,37 @@ class TurtleShell extends Card {
 	}
 
 	override onAttach(
-		_game: GameModel,
+		game: GameModel,
 		component: CardComponent,
 		observer: ObserverComponent,
 	) {
 		const {player} = component
-		let hasBeenActive = false
+		let activated = false
 
-		observer.subscribe(player.hooks.onTurnEnd, () => {
-			if (!component.slot.inRow()) return
-			if (player.activeRowEntity === component.slot.row.entity) {
-				hasBeenActive = true
-			}
-		})
+		observer.subscribe(
+			player.hooks.onActiveRowChange,
+			(_oldActiveHermit, newActiveHermit) => {
+				const hermitCard = game.components.find(
+					CardComponent,
+					query.card.isHermit,
+					query.card.row(query.row.hasCard(component.entity)),
+				)
+
+				if (hermitCard && newActiveHermit === hermitCard) {
```

Comment:
You should compare the card entities and not the objects. Unfortunately this code is wrong and technically works but i can't force it to error.

Reply:
Oops, fixed